crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2026,6 +2055,55 @@ impl<'db> Class<'db> {
         self.iter_mro(db).contains(&ClassBase::Class(other))
     }
 
+    /// Return the metaclass of this class.
+    pub(crate) fn metaclass(self, db: &'db dyn Db) -> Result<Type<'db>, MetaclassError> {
```

Comment:
We should make this a salsa query because what I understand is that this function can also be called cross module (e.g. resolve a base class from another file and we then lookup it's base). 

See https://github.com/astral-sh/ruff/pull/14087 for a more in depth explanation. 

```suggestion
    #[salsa::tracked]
    pub(crate) fn metaclass(self, db: &'db dyn Db) -> Result<Type<'db>, MetaclassError> {
```